### PR TITLE
test: add test for vlan.id - v8

### PR DESCRIPTION
**Pull request by AkakiAlice**
Ticket: https://github.com/OISF/suricata-verify/pull/1065

Description:
- Add Suricata-Verify test for vlan.id keyword

test.rules changes:
- Change ``vlan.id:!500;`` to ``vlan.id:!500,all;`` : [line 6](https://github.com/OISF/suricata-verify/pull/2194/commits/bac069336ca78e841213651d0ff4301d448cd987#diff-f2a2688beb95b96c6af4712f5bdc7f9d81e56e6ac25cb32a479e78930f825d89R6)
- Add rule to test ``count`` : [line 11](https://github.com/OISF/suricata-verify/pull/2194/commits/bac069336ca78e841213651d0ff4301d448cd987#diff-f2a2688beb95b96c6af4712f5bdc7f9d81e56e6ac25cb32a479e78930f825d89R11)

test.yaml changes:
- Add alert for new ``count`` rule : [line 62](https://github.com/OISF/suricata-verify/pull/2194/commits/bac069336ca78e841213651d0ff4301d448cd987#diff-29a94a3971562711c3f7b8b17857d5e388e81949dd2bd35a54a0c79141e70efdR62)

Redmine ticket: https://redmine.openinfosecfoundation.org/issues/1065

Previous PR: https://github.com/OISF/suricata-verify/pull/2188
Suricata PR: https://github.com/OISF/suricata/pull/12301

### CODE REVIEW

File: tests/detect-vlan-id/test.rules [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+alert ip any any -> any any (msg:"One Vlan ID is equal to 300"; vlan.id:300; sid:2;)
+alert ip any any -> any any (msg:"Last Vlan ID is equal to 400"; vlan.id:400,-1; sid:3;)
+alert ip any any -> any any (msg:"Vlan ID is equal to 300 with specific layer"; vlan.id:0x12C,1; sid:4;)
+alert ip any any -> any any (msg:"Vlan ID at layer 1 is not equal to 200"; vlan.id:!200,1; sid:5;)
+alert ip any any -> any any (msg:"There is no VLAN ID equal to 500"; vlan.id:!500,all; sid:6;)
+alert ip any any -> any any (msg:"VLAN ID at layer 2 is between 100 and 600"; vlan.id:100-600,2; sid:7;)
+alert ip any any -> any any (msg:"VLAN ID at layer 1 is less than 400"; vlan.id:<400,1; sid:8;)
+alert ip any any -> any any (msg:"One Vlan ID is greater than or equal to 200"; vlan.id:>=0xC8; sid:9;)
+alert ip any any -> any any (msg:"All the Vlan IDs are greater than 100"; vlan.id:>100,all; sid:10;)
+alert ip any any -> any any (msg:"Packet has 3 VLAN layers"; vlan.id:3,count; sid:11;)
```
Comment (catenacyber) [inline tests/detect-vlan-id/test.rules:11]: This PR is really fine.

You could also add `prefilter` in some signatures, like `vlan.id:>100,all; prefilter;`

You could also add a signature with `vlan.id:0,count;` and a packet without vlan

**Review by catenacyber — APPROVED** [review]
Good for me even if we can always test more

File: tests/detect-vlan-id/writepcap.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#! /usr/bin/env python3
+from scapy.all import *
```
Comment (jufajardini) [inline tests/detect-vlan-id/writepcap.py:2]: Can we also have the scapy version as a comment here in the script? Makes it easier to not lose the information :)

File: tests/detect-vlan-id/README.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+Test for checking the working of vlan.id keyword by creating rules and matching a crafted packet against them. The packet is an ICMP packet with 3 different VLAN ids [200,300,400].
```
Comment (jufajardini) [inline tests/detect-vlan-id/README.md:1]: As with the docs, these look better if we keep lines to a certain character count limit.
Let's try around 79 or 80?

File: tests/detect-vlan-id/README.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
...
+
+PCAP created with scapy 2.5.0.
```
Comment (jufajardini) [inline tests/detect-vlan-id/README.md:3]: Could you add the redmine ticket link reference here?

**Review by jufajardini — CHANGES_REQUESTED** [review]
As Philippe mentioned, very good work 👏🏽 

I do think we should take advantage of this moment to add the rule checks that we think would make the test more complete (the ones suggested by Philippe).

My comments are just to refine an already great job :)

**Pull request closed without merging**